### PR TITLE
Show dialogs on UI thread

### PR DESCRIPTION
**Pull request by lfkellogg**

### CODE REVIEW

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java [hunk 1]
```diff
@@ -215,34 +215,40 @@ public UpdateTask updateIfNewReleaseAvailable() {
...
-            showSignInDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
-
-    signInConfirmationDialog.show();
-
+    // We may not be on the main (UI) thread in some cases, specifically if the developer calls
+    // the basic config from the background. If we are already on the main thread, this will
+    // execute immediately.
+    hostActivity.runOnUiThread(
```
Comment (mannyjimenez0810) [inline firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java:223]: Can we do the work of creating the dialog not on the UI thread? Can we just have the showing of the dialog on the UI thread?

File: firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java [hunk 2]
```diff
@@ -247,6 +251,19 @@ public void updateToNewRelease_whenNewAabReleaseAvailable_showsUpdateDialog() {
         shadowOf(dialog).getMessage().toString());
   }
 
+  @Test
+  public void updateToNewRelease_fromABackgroundThread_showsUpdateDialog()
```
Comment (rachaprince) [inline firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java:255]: (Nit/ No action required) I'm just now noticing that some of these tests use the old name of `updateToNewRelease`, and some use the new name `updateIfNewReleaseAvailable` X)

**Review by rachaprince — APPROVED** [review]

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java (hunk 1 above)
Comment (lfkellogg) [reply firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java:223]: Good question - we actually can't. `new AlertDialog.Builder(hostActivity).create()` has to be called on the UI thread. I at first just wrapped the `.show()` call and then had to figure out why it was still broken :)

File: firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java (hunk 2 above)
Comment (lfkellogg) [reply firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java]: Ah, good catch. Fixed!

**Review by mannyjimenez0810 — APPROVED** [review]

**Pull request merged**